web/frontend/src/components/slam-2d-render.vue

```diff
@@ -63,25 +92,87 @@ controls.enableRotate = false;
 
 const raycaster = new MouseRaycaster({ camera, renderer, recursive: false });
 
-raycaster.on('click', (event) => {
+raycaster.on('click', (event: THREE.Event) => {
   const [intersection] = event.intersections as THREE.Intersection[];
   if (intersection && intersection.point) {
-    console.log(intersection.point);
+    emit('click', intersection.point);
   }
 });
 
-const markerSize = 0.5;
-const marker = new THREE.Mesh(
-  new THREE.PlaneGeometry(markerSize, markerSize).rotateX(-Math.PI / 2),
-  new THREE.MeshBasicMaterial({ color: 'red' })
-);
-marker.name = 'Marker';
-// This ensures the robot marker renders on top of the pointcloud data
-marker.renderOrder = 999;
+/*
+ * svgLoader example for webgl:
+ * https://github.com/mrdoob/three.js/blob/master/examples/webgl_loader_svg.html
+ */
+//
+const svgLoader = new SVGLoader(); 
+const makeMarker = async (url : string, name: string, scalar: number) => {
+  const data = await svgLoader.loadAsync(url);
+
+  const { paths } = data!;
+
+  const group = new THREE.Group();
+  group.scale.multiplyScalar(scalar);
+
+  for (const path of paths) {
+
+    const fillColor = path!.userData!.style.fill;
+
+    if (fillColor !== undefined && fillColor !== 'none') {
+      const material = new THREE.MeshBasicMaterial({
+        color: new THREE.Color().setStyle(fillColor)
+          .convertSRGBToLinear(),
+        opacity: path!.userData!.style.fillOpacity,
+        transparent: true,
+        side: THREE.DoubleSide,
+        depthWrite: false,
+        wireframe: false,
+      });
+
+      const shapes = SVGLoader.createShapes(path!);
+
+      for (const shape of shapes) {
+
+        const geometry = new THREE.ShapeGeometry(shape);
+        const mesh = new THREE.Mesh(geometry.rotateX(-Math.PI / 2).rotateY(Math.PI), material);
+        group.add(mesh);
+
+      }
+
+    }
+
+    const strokeColor = path!.userData!.style.stroke;
+
+    if (strokeColor !== undefined && strokeColor !== 'none') {
+
+      const material = new THREE.MeshBasicMaterial({
+        color: new THREE.Color().setStyle(strokeColor)
+          .convertSRGBToLinear(),
+        opacity: path!.userData!.style.strokeOpacity,
+        transparent: true,
+        side: THREE.DoubleSide,
+        depthWrite: false,
+        wireframe: false,
+      });
+
+      for (const subPath of path!.subPaths) {
+        const geometry = SVGLoader.pointsToStroke(subPath.getPoints(), path!.userData!.style);
+        if (geometry) {
+          const mesh = new THREE.Mesh(geometry.rotateX(-Math.PI / 2).rotateY(Math.PI), material);
```

Comment:
I see us using `-Math.PI / 2` and `Math.PI / 2` in many places but don't really understand why it is needed.
If we are doing it as a rotation, would it make sense to put those two values into constants at the top of the file with names (and maybe comments) that describe why they are needed so all the places performing transforms are clear?

Reply:
we always have to do `.rotateY(...)` since threeJS land is Y up.

when making svg markers we do `rotateX(...) to rotate the marker such that it is facing in the right direction.

both of these result from threeJS land specifics with how they organize their axes and directions of the axes...

we always rotate X by `-Math.PI / 2` but we do not have a constant for rotating around Y. 

I think putting the rotation values into constants would not add much, and could possible confuse.